src/libstd/fs/mod.rs

```diff
@@ -391,35 +484,85 @@ impl AsInnerMut<fs_imp::OpenOptions> for OpenOptions {
 
 impl Metadata {
     /// Returns whether this metadata is for a directory.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use std::fs::File;
+    ///
+    /// # fn foo() -> std::io::Result<()> {
+    /// let mut f = try!(File::create("foo.txt"));
+    /// let metadata = try!(f.metadata());
```

Comment:
I think a more conventional method of acquiring the metadata for a file is through the `fs::metadata` function instead of the `metadata` method on files maybe?